typed-racket-lib/typed-racket/types/abbrev.rkt

```diff
@@ -27,6 +27,7 @@
 
 ;; Convenient constructors
 (define -App make-App)
+(define -Self (make-F 'self-))
 (define -mpair make-MPair)
```

Comment:
Are we sure nothing bad is going to happen by hand-inserting in the symbol `'self-` here? I can't think of a specific concern, but it just seems a little funny to have that "magic symbol" there.

Reply:
@pnwamk, I can't come up with a better solution to representing `self` as a hygienic singleton type var. 
Also, it looks like I have to put it elsewhere since it is not a constructor.